lib/ansible/modules/identity/cyberark/cyberark_user.py

```diff
@@ -0,0 +1,461 @@
+            headers=headers,
+            status_code=http_exception.code)
+
+    except Exception as unknown_exception:
+
+        module.fail_json(
+            msg=("Unknown error while performing userAddOrUpdate."
+                 "\n*** end_point=%s%s\n%s" % (api_base_url, end_point, to_text(unknown_exception))),
+            payload=payload,
+            headers=headers,
+            status_code=-1)
+
+
+def userDelete(module):
+
+    # Get username from module parameters, and api base url
+    # along with validate_certs from the cyberark_session established
+    username = module.params["username"]
+    cyberark_session = module.params["cyberark_session"]
+    api_base_url = cyberark_session["api_base_url"]
+    validate_certs = cyberark_session["validate_certs"]
+
+    if module.check_mode:
+        # return if case of check_mode
+        module.exit_json(change=False)
+
+    # Prepare result, end_point, and headers
+    result = {}
+    end_point = "/PasswordVault/WebServices/PIMServices.svc/Users/{0}".format(
+        username)
+
+    headers = {'Content-Type': 'application/json'}
+    headers["Authorization"] = cyberark_session["token"]
+
+    try:
+
+        # execute REST action
+        response = open_url(
+            api_base_url + end_point,
+            method="DELETE",
+            headers=headers,
+            validate_certs=validate_certs)
+
+        result = {"result": {}}
+
+        return (True, result, response.getcode())
+
+    except (HTTPError, httplib.HTTPException) as http_exception:
+
+        exception_text = to_text(http_exception)
+        if http_exception.code == 404 and "ITATS003E" in exception_text:
+            # User does not exist
+            result = {"result": {}}
+            return (False, result, http_exception.code)
+        else:
+            module.fail_json(
+                msg=("Error while performing userDelete."
+                     "Please validate parameters provided."
+                     "\n*** end_point=%s%s\n ==> %s" % (api_base_url, end_point, exception_text)),
+                headers=headers,
+                status_code=http_exception.code)
+
+    except Exception as unknown_exception:
+
+        module.fail_json(
+            msg=("Unknown error while performing userDelete."
+                 "\n*** end_point=%s%s\n%s" % (api_base_url, end_point, to_text(unknown_exception))),
+            headers=headers,
+            status_code=-1)
+
+
+def userAddToGroup(module):
+
+    # Get username, and groupname from module parameters, and api base url
+    # along with validate_certs from the cyberark_session established
+    username = module.params["username"]
+    group_name = module.params["group_name"]
+    cyberark_session = module.params["cyberark_session"]
+    api_base_url = cyberark_session["api_base_url"]
+    validate_certs = cyberark_session["validate_certs"]
+
+    if module.check_mode:
+        # return if case of check_mode
+        module.exit_json(change=False)
```

Comment:
If we don't know whether a change would be made or not, we should always return changed=True.  It's better for a user to run their playbook unnecessarily than for a user to miss out on running a playbook to make changes because check_mode reported False.